scripts/user_retirement/utils/edx_api.py

```diff
@@ -474,28 +474,6 @@ def replace_usernames(self, username_mappings):
         return self._request('POST', api_url, json=data)
 
 
-class DemographicsApi(BaseApiClient):
```

Comment:
This particular endpoint wasn't part of the Demographics app in edx-platform... wouldn't we still need to support retiring learners in the Demographics IDA itself?

I don't know enough about the inner workings of the retirement pipeline yet to know if it's using this particular piece of code. It appears that the DemographicsAPI is still part of the (forked) `tubular` repo in our GitHub org... so maybe we're okay removing this one? 

Do you have a better understanding of the spaghetti?

Reply:
Aren't we about to retire the whole IDA, though? Does it matter if we miss a couple of retirement requests in the window before we delete all the data?

Reply:
on second thought, Justin is right, the DEPR of the IDA is soon but it's not _imminent_, better safe than sorry.